src/dialects/builtin/op_interfaces.rs

```diff
@@ -185,3 +185,28 @@ pub trait OneResultInterface: Op {
         Ok(())
     }
 }
+
+/// An [Op] that calls a function
+pub trait CallOpInterface {
```

Comment:
```suggestion
pub trait CallOpInterface : Op{
```

Also all `Op` interfaces must have a `fn verify(op: &dyn Op, ctx: &Context) -> Result<(), CompilerError>` member. Otherwise `impl_op_interface` won't work.

Reply:
Done.

Reply:
Did you push this commit? I don't see this in the PR diff

Reply:
Yep. Here it is - https://github.com/vaivaswatha/pliron/pull/8/commits/eb5a24dcd20d29d13d2ded57a9f060535c48dfeb